sound/soc/sof/intel/cnl.c

```diff
@@ -251,6 +251,7 @@ const struct snd_sof_dsp_ops sof_cnl_ops = {
 
 	/* ipc */
```

Comment:
On the commit message: "In order to prevent this race, re-schedule the d0i3_work as soon
as the SOF_CTRL_EVENT_KD event notification is received."

This will not prevet the race as D0i3x_work could be triggered just on the same moment as when FW is sending the KD notification, so this cannot be claimed in the commit message. There has to be a lower level mechanism to handle the race, similar to https://github.com/thesofproject/linux/pull/2663 ), to make sure the system handles this rare but possible scenario of timing.

Having said that, receiving a FW initiated IPC is clearly sign of device being not idle, so this patch still makes sense. It just doesn't fix the race.